server/channels/app/platform/websocket_reliable.go

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package platform
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+
+	"github.com/mattermost/mattermost/server/public/model"
+)
+
+func (ps *PlatformService) GetWSQueues(userID, connectionID string, seqNum int64) (*model.WSQueues, error) {
+	hub := ps.GetHubForUserId(userID)
+	if hub == nil {
+		return nil, nil
+	}
+	connRes := hub.CheckConn(userID, connectionID)
+	if connRes == nil {
+		return nil, nil
+	}
+	aq := connRes.ActiveQueue
+	dq := connRes.DeadQueue
+	dqPtr := connRes.DeadQueuePointer
+
+	// Nothing was written on this server. Early return.
+	if dq[0] == nil {
+		return nil, nil
+	}
+
+	// If seq-1 == last value in the dq.
+	if perfectMatch := !_hasMsgLoss(dq, dqPtr, seqNum); perfectMatch {
+		aqSlice, err := ps.getAQ(aq, connectionID, userID)
+		defer close(aq)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get from active queue: %w", err)
+		}
+		// send only aq
+		return &model.WSQueues{
+			ActiveQ:    aqSlice,
+			ReuseCount: connRes.ReuseCount,
+		}, nil
+	}
+
+	// If seq is there somewhere else in the dq.
```

Comment:
This comment is a little confusing, imo.

Reply:
hmm .. what would you rather say?

Reply:
That's the thing -- I'm not really sure what the comment means. :)

Reply:
Oh I see. Well, this is assuming that you understand how the active queue and dead queue works. At a high level, if everything is fine, then seq-1 is supposed to be the last entry in the dead queue pointed by deadQueuePointer. But if that's not the case, that means there's some msgs which are lost, so we check if "seq_num is somewhere else in the dead queue". Does that help?

Reply:
Ah, `check if seq_num is somewhere else in the dead queue` is understandable. I just had trouble parsing `If seq is there somewhere else in the dq.` Maybe consider changing?
```suggestion
	// Check if seq_num is somewhere else in the dead queue
```

Reply:
Yep. Can do.